src/main/java/com/minecolonies/coremod/util/StructureWrapper.java

```diff
@@ -196,9 +210,10 @@ private boolean checkForFreeSpace(@NotNull final BlockPos pos)
     /**
      * Place a structure into the world.
      *
-     * @param pos coordinates
+     * @param pos       coordinates
+     * @param complete  paste it complete (with structure blocks) or without
      */
-    private void placeStructure(@NotNull final BlockPos pos)
+    private void placeStructure(@NotNull final BlockPos pos, final boolean complete)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "placeStructure" is 17 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 41 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)